homeassistant/components/pcf8574/switch.py

```diff
@@ -0,0 +1,136 @@
+"""Support for switch sensor using I2C PCF8574 chip."""
+from pcf8574 import PCF8574
+import voluptuous as vol
+
+from homeassistant.components.switch import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME, DEVICE_DEFAULT_NAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import ToggleEntity
+
+from .const import DOMAIN
+
+CONF_I2C_PORT_NUM = "i2c_port_num"
+CONF_I2C_ADDRESS = "i2c_address"
+CONF_PINS = "pins"
+CONF_INVERT_LOGIC = "invert_logic"
+
+DEFAULT_NAME = "PCF8574 Switch"
+DEFAULT_I2C_ADDRESS = 0x20
+DEFAULT_I2C_PORT_NUM = 0
+DEFAULT_INVERT_LOGIC = False
+
+_SWITCHES_SCHEMA = vol.Schema({cv.positive_int: cv.string})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
New integrations should not offer a YAML configuration possibility, just a Config Flow. See https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision.

>Integrations that communicate with devices and/or services are configured via a config flow. In rare cases, we can make an exception.

And if you would need/get this exception, the way you currently set it up is not allowed due to ADR 007. See https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision.

Reply:
thanks for pointing out. removed all YAML configuration code